packages/ripple/src/index.ts

```diff
@@ -0,0 +1,134 @@
+import { getRippleGlobalNamespace } from './getRippleGlobalNamespace';
+import { Variant, colorMap, Mode } from './utils';
+
+const TRANSITION_TIME = 300;
+const RIPPLE_NAMESPACE = getRippleGlobalNamespace();
+
+interface Options {
+  variant: Variant;
+  darkMode: boolean;
+}
+
+let globalOptions: Options = { darkMode: false, variant: Variant.Default };
+
+function rippleEvent(event: MouseEvent) {
+  if (
+    // @ts-expect-error using HTMLElements to index as it provides a faster lookup when deciding if we should create a ripple effect on a given element
+    RIPPLE_NAMESPACE.registeredRippleElements[event.target]
+  ) {
+    createRippleEffect(event);
+  }
+}
+
+export function registerRipple(node: HTMLElement, options: Options) {
+  if (!RIPPLE_NAMESPACE) {
+    return;
+  }
+
+  globalOptions = options;
+
+  // @ts-expect-error using HTMLElements to index as it provides a faster lookup when deciding if we should create a ripple effect on a given element
+  RIPPLE_NAMESPACE.registeredRippleElements[node] = options;
+
+  if (!RIPPLE_NAMESPACE.setRippleListener) {
+    document.addEventListener('click', rippleEvent, { passive: true });
+
+    const styles = document.createElement('style');
+    styles.innerHTML = staticRippleStyles;
+    document.head.append(styles);
+
+    RIPPLE_NAMESPACE.setRippleListener = true;
+  }
+}
+
+export function unregisterRipple(node: HTMLElement) {
```

Comment:
Alternatively can just return this function from `registerRipple` so people can easily remove them. Similar to useEffect behavior, for easy cleanup.

Reply:
👍